astropy/utils/masked/function_helpers.py

```diff
@@ -533,7 +535,7 @@ def block(arrays):
 
 
 @dispatched_function
-def broadcast_arrays(*args, subok=True):
+def broadcast_arrays(*args, subok=False):
```

Comment:
Looking at the docstring, there seems to be a chance this may be regarded as an undesired API change.
However, I didn't see any test failing after the change so I have doubts. To be discussed !

Reply:
Had  a quick look. I think this change is correct - I'm fairly sure I used `subok=True` before because I wanted to keep the arrays being `Masked` - but that is done always anyway. Given that, probably best if the behaviour otherwise is just like it would be on an unmasked array, which indeed has default `subok=False` (I wish it hadn't but the conclusion when I implemented it was that the default should preserve the existing behaviour - which of course was reasonable).

Reply:
thanks for your feedback !